### PR TITLE
Fix `current_effective_deadline()`'s `-inf` disparity on `asyncio`

### LINKED ISSUE #484: `current_effective_deadline()` doesn't return `-inf` on `asyncio`
For example:

```python
import anyio
from anyio import CancelScope


async def main() -> None:
    with CancelScope() as cancel_scope:
        cancel_scope.cancel()
        print(anyio.current_effective_deadline())


anyio.run(main, backend="asyncio")  # prints inf
anyio.run(main, backend="trio")  # prints -inf
```

I *think* this discrepancy between backends is unintended—it violates the docstring of `AsyncBackend.current_effective_deadline()`.

### PR DESCRIPTION
**Pull request by gschaffner**
Closes #484.

### CODE REVIEW

**Review by agronholm — CHANGES_REQUESTED** [review]
Looks good, but could you add a note to the changelog too?

**Comment by gschaffner** [comment]
Yeah, done!

File: docs/versionhistory.rst [hunk 1]
```diff
@@ -38,6 +38,7 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 - Fixed ``start_blocking_portal()`` raising an unwarranted
   ``RuntimeError: This portal is not running`` if a task raises an exception that causes
   the event loop to be closed
+- Fixed ``current_effective_deadline()`` not returning ``-inf`` on asyncio (PR by Ganden Schaffner)
```
Comment (agronholm) [inline docs/versionhistory.rst:41]:
```suggestion
- Fixed ``current_effective_deadline()`` not returning ``-inf`` on asyncio when the
  currently active cancel scope has been cancelled (PR by Ganden Schaffner)
```

**Review by agronholm — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks!

**Pull request merged**